tests/test_registry.py

```diff
@@ -103,3 +103,17 @@ def test_multiple_connections(self, portals, dbus_con):
         app_id2 = mock_intf2.GetSessionAppId(session2.handle)
         assert app_id2 == expected_app_id
         dbus_con2.close()
+
+    def test_no_reregister(self, portals, dbus_con):
+        registry_intf = xdp.get_portal_iface(dbus_con, "Registry", domain="host")
+        mock_intf = xdp.get_mock_iface(dbus_con)
+
+        expected_app_id = "org.example.CorrectAppId"
+
+        registry_intf.Register(expected_app_id, {})
+        session = self.create_dummy_session(dbus_con)
+        app_id = mock_intf.GetSessionAppId(session.handle)
+        assert app_id == expected_app_id
+
+        with pytest.raises(dbus.exceptions.DBusException):
```

Comment:
All of  those should probably test against a more specific error. a timeout for example shouldn't make this succeed.

Reply:
Indeed, they now do that.